doc/fork_support.md

```diff
@@ -0,0 +1,49 @@
+# Background #
+
+In Python multithreading is ineffective at concurrency for CPU bound tasks
+due to the GIL.
+(global interpreter lock).  Users use multiprocessing, subprocess, 
+and concurrent.futures.ProcessPoolExecutor, to work around the GIL.
+These modules call fork() underneath the hood. Various issues have
+been reported when using these modules with gRPC Python.
```

Comment:
We could add some more context, something like..

gRPC core, which gRPC python wraps, uses multi threading for performance, and hence does not support fork().

Reply:
Good idea, added that.